resolver/SuggestPackage.cc

```diff
@@ -304,6 +314,23 @@ bool SuggestPackage::tryPackageCorrections(core::Context ctx, core::ErrorBuilder
             for (auto match : missingExports) {
                 pkgCtx.addMissingExportSuggestions(e, match);
             }
+
+            if (auto fullConstant = ctx.locAt(unresolved.loc).source(ctx)) {
+                if (missingExports.size() == 1) {
+                    // if we have a single unambiguous export
+                    // suggestion, then we can use that information to
+                    // make the error message reference the exact
+                    // package we believe the constant came from
+                    auto missingExport = missingExports.front();
+                    auto pkgName = ctx.state.packageDB().getPackageInfo(missingExport.srcPkg).show(ctx);
+                    e.setHeader("Package `{}` does not export `{}`", pkgName, *fullConstant);
+                } else {
+                    // otherwise we should be explicit that it's
+                    // probably an export problem but be a bit cagey
+                    // about WHAT export problem it is
+                    e.setHeader("`{}` is not a public constant", *fullConstant);
```

Comment:
do you think it would be possible to somehow check whether the suggestion is like an "exact" match?

If I'm not mistaken, we're relying on "did you mean" suggestions here, which can suggest `SomeThing` if the user typed `Something`

(or is it more or less guaranteed that this code path only triggers for specifically packager related errors? maybe this is not worth optimizing for if trevor is going to be changing how this works shortly?)

Reply:
I _believe_ this should only happen for packager-related errors, although I think it's not impossible that I'm missing some path that confuses it. (I do also think Trevor's changes are going to make that belief ironclad, so while I'd be happy to do some derisking, I do think it's got diminishing returns especially if we think we can land the new package approach relatively quickly.)